src/core/ddsc/src/shm_monitor.c

```diff
@@ -44,12 +44,12 @@ void shm_monitor_init(shm_monitor_t* monitor)
 
 void shm_monitor_destroy(shm_monitor_t* monitor) 
 {
-    //note: we must ensure no readers are actively using the monitor anymore,    
     shm_monitor_wake_and_disable(monitor);
-    while(monitor->m_state == SHM_MONITOR_RUNNING); //spin until callbacks are no longer processed
+    // waiting for the readers to be detaced is not necessary, they will be detached when the listener is destroyed
+    // but if we for some reason need this we could do it like this 
+    // while(monitor->m_state == SHM_MONITOR_RUNNING || monitor->m_number_of_attached_readers > 0);
+    while(monitor->m_state == SHM_MONITOR_RUNNING);
```

Comment:
This will cause this thread to spin at 100% CPU usage, is this required due to timing?
Maybe adding even a small (1 millisecond) sleep will reduce the cpu overhead by a large amount, while still keeping a quick cleanup.

Reply:
I will look into it, it should suffice to wait for the thread to join.